client/src/components/Home.js

```diff
@@ -1,30 +1,98 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import Button from '@material-ui/core/Button';
 import Icon from '@material-ui/core/Icon';
 import Grid from '@material-ui/core/Grid';
 import { withStyles } from '@material-ui/core/styles';
 import Leaderboard from './Leaderboard';
+import LoaderComponent from './common/loaderComponent/loaderComponent';
+
+import socket from '../api/socketService';
 
 const styles = theme => ({
   button: { margin: theme.spacing.unit },
   rightIcon: { marginLeft: theme.spacing.unit },
 });
 
-const Home = ({ classes }) => (
-  <React.Fragment>
-    <Grid container direction="row" justify="center" alignItems="center">
-      <Button className={classes.button} variant="contained" color="primary" size="large">
-        Play
-        <Icon className={classes.rightIcon}>send</Icon>
-      </Button>
-      <Button className={classes.button} variant="contained" color="secondary" size="large">
-        Find a Friend
-      </Button>
-    </Grid>
-    <Leaderboard limit={5} />
-  </React.Fragment>
-);
+class Home extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+      loaderMessage: 'waiting for another player to join..',
+      redirectToGame: false,
+    };
+    this.playGame = this.playGame.bind(this);
+    this.renderRedirectToGame = this.renderRedirectToGame.bind(this);
+    this.toBeRenderJSX = this.toBeRenderJSX.bind(this);
+  }
+
+  componentDidMount() {
+    socket.emit('connection', () => {
+      // eslint-disable-next-line no-console
+      console.log('Socket connection is set with back-end server.');
+    });
+
+    socket.on('getReady', (message) => {
+      // eslint-disable-next-line no-console
+      console.log(message);
+      const loaderMessage = 'Wooh! Found a player, get ready. you have 3 seconds.';
+      this.setState({ loaderMessage });
+      setTimeout(() => this.setState({ redirectToGame: true }), 2500);
+    });
+  }
+
+  playGame() {
+    this.setState({ loading: true });
+    socket.emit('joinGame', 'nothing');
+    socket.on('assignPlayerId', (playerId) => {
+      // eslint-disable-next-line no-console
+      console.log('playerId assigned: ', playerId);
+      this.state.playerId = playerId;
```

Comment:
No `setState`?

Reply:
This should've been caught by the pre-commit hook, but I guess it didn't work. 

I'll add `yarn lint` in travis also so this kind of errors are caught before PR is submitted for review.